aggregator_core/src/datastore.rs

```diff
@@ -1574,7 +1578,7 @@ impl<C: Clock> Transaction<'_, C> {
                     .prepare_cached(
                         "DELETE FROM client_reports
                         USING tasks
-                        WHERE client_reports.id = tasks.task_id
+                        WHERE client_reports.task_id = tasks.id
```

Comment:
Was this just broken and we didn't notice because nothing was exercising garbage collection?

Reply:
It was broken, but not because GC wasn't exercised -- this query was silently not deleting anything. But this wasn't noticed because this codepath is only taken if the attempted write is already expired; and in that case the read methods would treat the row as nonexistent. So this problem was undetectable without doing something like a custom SQL query to read the row disregarding expiration status.

The practical upshot of this was small -- a row would be written (and correctly ignored as garbage) and deleted once the garbage collector got to it, instead of no row being written.

If we want to do better, I think there are a few options:
(a) write some custom SQL / a `test-util`-only method that reads client report rows regardless of expiration status.
(b) drop this codepath entirely -- writing an expired entity will actually write a row to the DB, and we lean on the reader's expiration check to act as if the row is not there.

Thoughts?

Reply:
Oh, I got mixed up: I thought the problem was that the query referred to columns on `client_reports` and `tasks` that don't exist, and was surprised that it wasn't straightforwardly rejected by Postgres. I understand the problem better now.

I'm not sure how much we should invest in detailed testing on this. It sounds like despite this bug, all the external behavior was correct: the row in question would get ignored by most queries and eventually would get deleted by GC. I'm not sure if the extra test code and the extra coupling to the code under test would be worth the assurance that the row actually gets deleted.

I think you understand this code much better than I do now so I'll defer to your judgment on the practicality of testing this.